File: eis2/src/app/service/report-time.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { DialogueVo } from 'src/app/data/dialogue/dialoguevo';
import { HttpParams, HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class ReportTimeService {
  constructor(private http: HttpClient) { }

  generateOF288TimeInvoiceReport(timeReportFilter): Observable<DialogueVo> {
    return this.http.post<DialogueVo>('/service/reports/time/of-288', timeReportFilter);
  }

  generateOF286TimeInvoiceReport(of286TimeInvoiceReportFilter): Observable<DialogueVo> {
    return this.http.post<DialogueVo>('/service/reports/time/of-286', of286TimeInvoiceReportFilter);
  }

  generateShiftsInExcessOfStandardHoursReport(shiftsInExcessOfStandardHoursReportFilter): Observable<DialogueVo> {
    return this.http.post<DialogueVo>('/service/reports/time/shifts-in-excess-of-standard-hours', shiftsInExcessOfStandardHoursReportFilter);
  }

  generatePersonnelTimeReport(personnelTimeReportFilter): Observable<DialogueVo> {
    return this.http.post<DialogueVo>('/service/reports/time/personnel', personnelTimeReportFilter);
  }

  generateWorkRestRatioReport(of286TimeInvoiceReportFilter): Observable<DialogueVo> {
    return this.http.post<DialogueVo>('/service/reports/time/work-rest-ratio', of286TimeInvoiceReportFilter);
  }

  generateSummaryHoursWorkedReport(summaryHoursWorkedReportFilter): Observable<DialogueVo> {
    return this.http.post<DialogueVo>('/service/reports/time/summary-hours-worked', summaryHoursWorkedReportFilter);
  }

  generateMissingDaysOfPostingsReport(missingDaysOfPostingsReportFilter): Observable<DialogueVo> {
    return this.http.post<DialogueVo>('/service/reports/time/missing-days-of-postings', missingDaysOfPostingsReportFilter);
  }

  generateCrewRosterReport(timeReportFilter): Observable<DialogueVo> {
    return this.http.post<DialogueVo>('/service/reports/time/crew-roster', timeReportFilter);
  }

  generateVendorResourceSummaryReport(vendorResourceSummaryReportFilter): Observable<DialogueVo> {
    return this.http.post<DialogueVo>('/service/reports/time/vendor-resource-summary',vendorResourceSummaryReportFilter);
  }

  getResourceListForSelectedIncident(listName, incidentId, incidentGroupId): Observable<DialogueVo> {
    const params: HttpParams = new HttpParams()
      .set('incidentId', incidentId)
      .set('incidentGroupId', incidentGroupId)
      .set('listName', listName);
    return this.http.get<DialogueVo>('/service/reports/time/get-resource-list-for-selected-incident', {params: params});
  }

  getRequestNumbersForSelectedIncident(incidentId, incidentGroupId): Observable<DialogueVo> {
    const params: HttpParams = new HttpParams()
      .set('incidentId', incidentId)
      .set('incidentGroupId', incidentGroupId);
    return this.http.get<DialogueVo>('/service/reports/time/get-request-numbers-for-selected-incident', {params: params});
  }

  getAgencyResourcesForSelectedIncident(incidentId, incidentGroupId): Observable<DialogueVo> {
    const params: HttpParams = new HttpParams()
      .set('incidentId', incidentId)
      .set('incidentGroupId', incidentGroupId);
    return this.http.get<DialogueVo>('/service/reports/time/get-agency-resources-for-selected-incident', {params: params});
  }

  getResourceInvoiceList(resourceId): Observable<DialogueVo> {
    const params: HttpParams = new HttpParams()
      .set('resourceId', resourceId)
    return this.http.get<DialogueVo>('/service/reports/time/resource-invoice-list', {params: params});
  }

  deleteLastTimeInvoice(timeReportFilter): Observable<DialogueVo> {
    return this.http.post<DialogueVo>('/service/reports/time/delete-last-invoice', timeReportFilter);
  }

  getResourceData(incidentId, incidentGroupId){
    const params: HttpParams = new HttpParams()
      .set('incidentId', incidentId)
      .set('incidentGroupId', incidentGroupId);
    return this.http.get<any>('/service/reports/time/get-resource-data', {params: params});
  }
}

File: isuite-core-ui/src/main/java/gov/nwcg/isuite/core/service/ReportTimeService.java
package gov.nwcg.isuite.core.service;

import java.util.Collection;

import gov.nwcg.isuite.core.reports.filter.MissingDaysOfPostingsReportFilter;
import gov.nwcg.isuite.core.reports.filter.OF286TimeInvoiceReportFilter;
import gov.nwcg.isuite.core.reports.filter.PersonnelTimeReportFilter;
import gov.nwcg.isuite.core.reports.filter.ShiftsInExcessOfStandardHoursReportFilter;
import gov.nwcg.isuite.core.reports.filter.VendorResourceSummaryReportFilter;
import gov.nwcg.isuite.core.reports.filter.SummaryHoursWorkedReportFilter;
import gov.nwcg.isuite.core.reports.filter.WorkRestRatioReportFilter;
import gov.nwcg.isuite.core.vo.PersonnelTimeRepCTDVo;
import gov.nwcg.isuite.core.vo.dialogue.DialogueVo;

import gov.nwcg.isuite.framework.exceptions.PersistenceException;
import gov.nwcg.isuite.framework.exceptions.ServiceException;
import gov.nwcg.isuite.framework.filter.TimeReportFilter;


public interface ReportTimeService {

	/**
	 * 
	 * @param filter
	 * @param incidentId
	 * @param incidentResourceId
	 * @return
	 * @throws ServiceException
	 * @throws PersistenceException 
	 */
	public DialogueVo generateOF288TimeInvoiceReport(TimeReportFilter filter, DialogueVo dialogueVo) throws ServiceException, PersistenceException;
	
	/**
	 * 
	 * @param filter
	 * @param incidentId
	 * @return
	 * @throws ServiceException
	 * @throws PersistenceException 
	 */
	public DialogueVo generateOF286TimeInvoiceReport(OF286TimeInvoiceReportFilter filter, DialogueVo dialogueVo) throws ServiceException, PersistenceException;

	public DialogueVo reprintOF286TimeInvoice(TimeReportFilter filter, DialogueVo dialogueVo) throws ServiceException, PersistenceException;
	
	public DialogueVo reprintOF288TimeInvoice(TimeReportFilter filter, DialogueVo dialogueVo) throws ServiceException, PersistenceException;
	
	public DialogueVo generateWorkRestRatioReport(WorkRestRatioReportFilter filter, DialogueVo dialogueVo) throws ServiceException, PersistenceException;
	
	public DialogueVo generateSummaryHoursWorkedReport(SummaryHoursWorkedReportFilter filter, DialogueVo dialogueVo) throws ServiceException, PersistenceException;
	
	public DialogueVo generateMissingDaysOfPostingsReport(MissingDaysOfPostingsReportFilter filter, DialogueVo dialogueVo) throws ServiceException, PersistenceException;
	
	public DialogueVo generateCrewRosterReport(TimeReportFilter filter, DialogueVo dialogueVo) throws ServiceException, PersistenceException;
	
	public DialogueVo generateVendorResourceSummaryReport(VendorResourceSummaryReportFilter filter, DialogueVo dialogueVo) throws ServiceException, PersistenceException;
	
	
	 /**
	 * 
	 * @param filter
	 * @param incidentId
	 * @param dialogueVo
	 * @return
	 * @throws ServiceException
	 */
	public DialogueVo generateShiftsInExcessOfStandardHoursReport(ShiftsInExcessOfStandardHoursReportFilter filter, DialogueVo dialogueVo) throws ServiceException, PersistenceException;
	
	DialogueVo getResourceListForSelectedIncident(String listName, Long incidentId, Long incidentGroupId, DialogueVo dialogueVo) throws ServiceException;
	
	DialogueVo getAgencyResourcesForSelectedIncident(Long incidentId, Long incidentGroupId, DialogueVo dialogueVo) throws ServiceException;
	
//	DialogueVo getRequestNumbersForSelectedIncident(Long incidentId, Long incidentGroupId, DialogueVo dialogueVo) throws ServiceException;
//	
//	DialogueVo getContractedRequestNumbersForSelectedIncident(Long incidentId, Long incidentGroupId, DialogueVo dialogueVo) throws ServiceException;
//	
//	DialogueVo getNonContractedRequestNumbersForSelectedIncident(Long incidentId, Long incidentGroupId, DialogueVo dialogueVo) throws ServiceException;
//	
//	DialogueVo getPersonRequestNumbersForSelectedIncident(Long incidentId, Long incidentGroupId, DialogueVo dialogueVo) throws ServiceException;
//	
//	DialogueVo getPersonResourceNamesForSelectedIncident(Long incidentId, Long incidentGroupId, DialogueVo dialogueVo) throws ServiceException;
//	
//	DialogueVo getContractedPersonResourceNamesForSelectedIncident(Long incidentId, Long incidentGroupId, DialogueVo dialogueVo) throws ServiceException;
//	
//	DialogueVo getNonContractedPersonResourceNamesForSelectedIncident(Long incidentId, Long incidentGroupId, DialogueVo dialogueVo) throws ServiceException;
//	
//	DialogueVo getCrewNamesForSelectedIncident(Long incidentId, Long incidentGroupId, DialogueVo dialogueVo) throws ServiceException;
	
	DialogueVo saveActualReleaseDateAndTime(TimeReportFilter filter, DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo getAgencyTreeDataForPersonnelTimeRep( Long incidentOrGroupId, Boolean isIncidentGroup, DialogueVo dialogueVo) throws ServiceException;

	public DialogueVo getResourceData(Long incidentId, Long incidentGroupId, DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo generatePersonnelTimeReport(PersonnelTimeReportFilter filter, DialogueVo dialogueVo) throws Exception;
	
	public DialogueVo getResourceInvoiceList(Long resourceId, DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo deleteLastTimeInvoice(TimeReportFilter filter, DialogueVo dialogueVo) throws ServiceException;
}
